Ensures that any directories specifed as part of ``outdir`` exist.

mabruzzo opened the pull request:
This is accomplished through the new ``Ensure_Outdir_Exists`` function.

Along the way, I introduced a helper function ``Is_Root_Proc()``, local to ``io.cpp`` so that we can reduce some of the ``ifdef`` statements. If you don't like this, I'm happy to remove it.

mabruzzo committed:
Ensures that any directories specifed as part of ``outdir`` exist.

This is accomplished through the new ``Ensure_Outdir_Exists`` function. Along the way, I introduced a helper function ``Is_Root_Proc()``, local to ``io.cpp`` so that we can reduce some of the ``ifdef`` statements.

alwinm commented on src/io/io.cpp:
```diff
@@ -6,9 +6,11 @@
 #include <fstream>
 #include <iostream>
 #include <string>
+#include <vector>
```
Basic question, what is <vector> providing in this PR?

mabruzzo replied:
that was an oversight. I thought I removed it. I'll fix it now

bcaddy commented on src/io/io.cpp:
```diff
@@ -64,11 +75,9 @@ void Create_Log_File(struct parameters P)
 {
-#ifdef MPI_CHOLLA
-  if (procID != 0) {
+  if (!!Is_Root_Proc()) {
```
Is the double negation intended? 

Also, several months ago there was a discussion on using keyword vs. character Boolean operators (! vs not, || vs or, etc). If I recall correctly we decided to use keyword operators.

mabruzzo replied:
The double negative was definitely an oversight. Thanks for catching that.

Gotcha, I'll shift to using `not` and `or`

mabruzzo committed:
address PR review comments from @alwinm & @bcaddy

- removed an unnecessary include-directive
- fixed a double-negative boolean negation operation (I previously wrote `!!` when it should have just been `!`)
- replaced `!` operator with `not` keyword and `||` operator with `or` keyword

evaneschneider committed:
Merge branch 'dev' into missing-outdir
